From 3d9ef2a95cea0a6097d25d5241e1d39e6f67857f Mon Sep 17 00:00:00 2001
From: Jensen Huang <jensenhuang@friendlyarm.com>
Date: Sun, 15 Oct 2023 12:04:23 +0800
Subject: [PATCH 1/2] libwebsockets: disable '-Wenum-int-mismatch' for gcc-13

Signed-off-by: Jensen Huang <jensenhuang@friendlyarm.com>
---
 libs/libwebsockets/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libs/libwebsockets/Makefile b/libs/libwebsockets/Makefile
index c71510b01..1e20189f8 100644
--- a/libs/libwebsockets/Makefile
+++ b/libs/libwebsockets/Makefile
@@ -28,6 +28,10 @@ CMAKE_INSTALL:=1
 include $(INCLUDE_DIR)/package.mk
 include $(INCLUDE_DIR)/cmake.mk
 
+ifeq ($(CONFIG_GCC_USE_VERSION_13),y)
+TARGET_CFLAGS += -Wno-error=enum-int-mismatch
+endif
+
 CMAKE_OPTIONS += -DLWS_IPV6=$(if $(CONFIG_IPV6),ON,OFF)
 CMAKE_OPTIONS += -DISABLE_WERROR=ON
 CMAKE_OPTIONS += -DCMAKE_BUILD_TYPE=Release
-- 
2.25.1

